### PR TITLE
Add vCPU hard pinning

### PR DESCRIPTION
**Pull request by basraayman**
Added the option to allow hard pinning to be configured.

### CODE REVIEW

**Comment by siddharth-nutanix** [comment]
/ok-to-test
>Acceptance test run status: success
>Line code coverage is 60.2

**Review by siddharth-nutanix — APPROVED** [review]

File: nutanix/resource_nutanix_virtual_machine.go [hunk 1]
```diff
@@ -1085,6 +1090,7 @@ func resourceNutanixVirtualMachineRead(d *schema.ResourceData, meta interface{})
 	}
 
 	d.Set("enable_cpu_passthrough", resp.Status.Resources.EnableCPUPassthrough)
+	d.Set("is_vcpu_hard_pinned", resp.Status.Resources.EnableCPUPinning)
```
Comment (siddharth-nutanix) [inline nutanix/resource_nutanix_virtual_machine.go:1093]:
```suggestion
	d.Set("is_vcpu_hard_pinned", utils.BoolValue(resp.Status.Resources.EnableCPUPinning))
```

Comment (siddharth-nutanix) [reply nutanix/resource_nutanix_virtual_machine.go:1093]: @basraayman Can you check the above suggestion ? 
I feel we need to convert boolean pointer to boolean while setting the value to resource data.

**Review by shreevari — APPROVED** [review]

**Review by siddharth-nutanix — CHANGES_REQUESTED** [review]

File: nutanix/resource_nutanix_virtual_machine.go (hunk 1 above)
Comment (siddharth-nutanix) [reply nutanix/resource_nutanix_virtual_machine.go:1093]: My bad values do get set with boolean pointers as well.

**Review by siddharth-nutanix — APPROVED** [review]

**Review by basraayman — COMMENTED** [review]
All good, thank you. :)

**Pull request merged**